### PR TITLE
Add an appendix about numeric strings

### PR DESCRIPTION
**Pull request by Girgias**
Add a dedicated appendix explaining numeric strings, drive-by fixes.

### CODE REVIEW

**Comment by cmb69** [comment]
Hmm, where is the appendix? :)

File: appendices/numeric-strings.xml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ </informalexample>
+
+ <para>
+  PHP also has a concept of <emphasis>leading</emphasis> numeric strings.
+  This is simply a string which starts like a numeric string followed by
+  any characters.
+ </para>
+
+ <section xml:id="numeric-string.conversion">
+  <title>Strings used in numeric contextes</title>
```
Comment (cmb69) [inline appendices/numeric-strings.xml:35]: Nit:
```suggestion
  <title>Strings used in numeric contexts</title>
```

File: appendices/numeric-strings.xml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      The <type>string</type> is not numeric, throw a
+      <classname>TypeError</classname>.
+     </simpara>
+    </listitem>
+   </orderedlist>
+  </para>
+ </section>
+
+ <section xml:id="numeric-string.prior">
+  <title>Behaviour prior to PHP 8.0.0</title>
```
Comment (cmb69) [inline appendices/numeric-strings.xml:73]: Nit:
```suggestion
  <title>Behavior prior to PHP 8.0.0</title>
```

**Review by cmb69 — COMMENTED** [review]
I wonder whether the numeric-strings.xml file should better go into language/types/?

**Comment by Girgias** [comment]
> 
> 
> I wonder whether the numeric-strings.xml file should better go into language/types/?

I was thinking about that but it's not really a type, so I thought an appendix might be better. But that's easily changeable.

**Comment by cmb69** [comment]
Well, neither declarations nor type-juggling are types. :)

**Comment by Girgias** [comment]
> 
> 
> Well, neither declarations nor type-juggling are types. :)

That's true... let me rework it

File: language/types/float.xml [hunk 3]
```diff
@@ -81,23 +81,35 @@ EXPONENT_DNUM (({LNUM} | {DNUM}) [eE][+-]? {LNUM})
...
+    For values of other types, the conversion is performed by converting the
+    value to <type>int</type> first and then to <type>float</type>. See
+    <link linkend="language.types.integer.casting">Converting to integer</link>
+    for more information.
    </para>
-  </note>
 
+   <note>
+    <para>
+     A notice is thrown if an <type>object</type> is converted to <type>float</type>.
```
Comment (Girgias) [inline language/types/float.xml:109]: I wonder why this note exits, as the same applies for integers but there it's just declared undefined behaviour (which would also apply for objects...)

Comment (cmb69) [reply language/types/float.xml:109]: See https://github.com/php/doc-en/pull/156/files#diff-4af4f41802cf4b045bb14a24dfac1bea5f6bf1682797f0bf22c98925b8af0a60L92-R99. Apparently, that was not the case in PHP 4; could probably drop that note.

File: language/types/type-juggling.xml [hunk 4]
```diff
@@ -74,8 +74,9 @@ examples:
 
  <note>
   <para>
-   The behaviour of an automatic conversion to <type>array</type> is currently
-   undefined.
+   An <type>array</type> cannot be coerced and will throw a
+   <classname>TypeError</classname> as of PHP 8.0.0.
+   Previously the behaviour was undefined.
```
Comment (Girgias) [inline language/types/type-juggling.xml:79]: Just realised that I probably misunderstood this note, probably should revert this.

**Pull request closed without merging**